`TestTransactionAggregation` systemtest is flaky

Observed in a few PRs:

```
 === RUN   TestTransactionAggregation
    aggregation_test.go:79:   map[string]any{
          	"events": []any{
          		map[string]any{
          			... // 8 identical entries
          			"processor": map[string]any{"event": string("metric"), "name": string("metric")},
          			"service":   map[string]any{"language": map[string]any{"name": string("go"), "version": string("2.0")}, "name": string("systemtest"), "node": map[string]any{"name": string("beowulf")}, "runtime": map[string]any{"name": string("gc"), "version": string("2.0")}},
          			"transaction": map[string]any{
          				"duration": map[string]any{
          					"histogram": map[string]any{
        - 						"counts": []any{float64(5)},
        + 						"counts": []any{float64(10)},
          						"values": []any{float64(1.003519e+06)},
          					},
          				},
        - 				"name": string("abc"),
        + 				"name": string("def"),
          				"root": bool(true),
          				"type": string("backend"),
          			},
          		},
          		map[string]any{
          			... // 8 identical entries
          			"processor": map[string]any{"event": string("metric"), "name": string("metric")},
          			"service":   map[string]any{"language": map[string]any{"name": string("go"), "version": string("2.0")}, "name": string("systemtest"), "node": map[string]any{"name": string("beowulf")}, "runtime": map[string]any{"name": string("gc"), "version": string("2.0")}},
          			"transaction": map[string]any{
          				"duration": map[string]any{
          					"histogram": map[string]any{
        - 						"counts": []any{float64(10)},
        + 						"counts": []any{float64(5)},
          						"values": []any{float64(1.003519e+06)},
          					},
          				},
        - 				"name": string("def"),
        + 				"name": string("abc"),
          				"root": bool(true),
          				"type": string("backend"),
          			},
          		},
          		map[string]any{"@timestamp": string("2021-09-15T20:11:06.000Z"), "agent": map[string]any{"name": string("elastic-node")}, "cloud": map[string]any{"account": map[string]any{"id": string("account_id"), "name": string("account_name")}, "availability_zone": string("cloud_availability_zone"), "machine": map[string]any{"type": string("machine_type")}, "project": map[string]any{"id": string("project_id"), "name": string("project_name")}, ...}, "container": map[string]any{"id": string("container-id")}, ...},
          	},
          }
        
        Run `make update check-approvals` to verify the diff
        
    server.go:132: log file: /var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-9578/src/github.com/elastic/apm-server/systemtest/logs/TestTransactionAggregation/apm-server
--- FAIL: TestTransactionAggregation (2.53s)

```

## Related

* https://github.com/elastic/apm-server/pull/9578#issuecomment-1317091185
* https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/main/930/testReport

--- a/systemtest/approvals.go
+++ b/systemtest/approvals.go
@@ -69,6 +69,7 @@ var apmEventSortFields = []string{
 	"transaction.id",
 	"span.id",
 	"error.id",
+	"transaction.name",
 	"span.destination.service.resource",
 	"transaction.type",
 	"span.type",

--- a/systemtest/approvals/TestIngestPipelineDataStreamMigration.approved.json
+++ b/systemtest/approvals/TestIngestPipelineDataStreamMigration.approved.json
@@ -264,6 +264,54 @@
             "agent": {
                 "name": "go"
             },
+            "data_stream": {
+                "dataset": "apm.internal",
+                "namespace": "migrated",
+                "type": "metrics"
+            },
+            "ecs": {
+                "version": "dynamic"
+            },
+            "event": {
+                "agent_id_status": "missing",
+                "ingested": "dynamic",
+                "outcome": "failure"
+            },
+            "metricset": {
+                "name": "service_destination"
+            },
+            "observer": {
+                "hostname": "dynamic",
+                "name": "instance-0000000000",
+                "type": "apm-server",
+                "version": "dynamic"
+            },
+            "processor": {
+                "event": "metric",
+                "name": "metric"
+            },
+            "service": {
+                "name": "main"
+            },
+            "span": {
+                "destination": {
+                    "service": {
+                        "resource": "dest_resource",
+                        "response_time": {
+                            "count": 1,
+                            "sum": {
+                                "us": 10220
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        {
+            "@timestamp": "2022-09-12T03:52:00.000Z",
+            "agent": {
+                "name": "go"
+            },
             "container": {
                 "id": "spawn-6cd7c2e1-43d5-4c89-8480-e178b3bc70d1"
             },
@@ -475,54 +523,6 @@
             "transaction": {
                 "name": "name",
                 "type": "type"
-            }
-        },
-        {
-            "@timestamp": "2022-09-12T03:52:00.000Z",
-            "agent": {
-                "name": "go"
-            },
-            "data_stream": {
-                "dataset": "apm.internal",
-                "namespace": "migrated",
-                "type": "metrics"
-            },
-            "ecs": {
-                "version": "dynamic"
-            },
-            "event": {
-                "agent_id_status": "missing",
-                "ingested": "dynamic",
-                "outcome": "failure"
-            },
-            "metricset": {
-                "name": "service_destination"
-            },
-            "observer": {
-                "hostname": "dynamic",
-                "name": "instance-0000000000",
-                "type": "apm-server",
-                "version": "dynamic"
-            },
-            "processor": {
-                "event": "metric",
-                "name": "metric"
-            },
-            "service": {
-                "name": "main"
-            },
-            "span": {
-                "destination": {
-                    "service": {
-                        "resource": "dest_resource",
-                        "response_time": {
-                            "count": 1,
-                            "sum": {
-                                "us": 10220
-                            }
-                        }
-                    }
-                }
             }
         },
         {

